onnxmltools/convert/keras/operator_converters/Bidirectional.py

```diff
@@ -0,0 +1,251 @@
+    container.add_initializer(tensor_r_name, onnx_proto.TensorProto.FLOAT,
+                              [2, 4 * hidden_size, hidden_size], np.concatenate([W_h, W_h_back]).flatten())
+    lstm_input_names.append(tensor_r_name)
+
+    # Add bias vectors at different places in the original LSTM if needed
+    if b is not None:
+        tensor_b_name = scope.get_unique_variable_name(operator.full_name + '_B')
+        container.add_initializer(tensor_b_name, onnx_proto.TensorProto.FLOAT, [2, 8 * hidden_size],
+                                  np.concatenate([b, b_back]).flatten())
+        lstm_input_names.append(tensor_b_name)
+    else:
+        lstm_input_names.append('')  # the name of a non-existing optional variable is an empty string
+
+    # sequence_lens, this input is not used when converting Keras Bidirectional.
+    lstm_input_names.append('')
+
+    # need the zero initializer to correct some engine shape inference bug.
+    state_shape = (2, 1, hidden_size)
+    initial_h_name = scope.get_unique_variable_name(operator.full_name + '_initial_h')
+    container.add_initializer(initial_h_name, onnx_proto.TensorProto.FLOAT, state_shape,
+                              np.zeros(shape=state_shape).flatten())
+    lstm_input_names.append(initial_h_name)
+    initial_c_name = scope.get_unique_variable_name(operator.full_name + '_initial_c')
+    container.add_initializer(initial_c_name, onnx_proto.TensorProto.FLOAT, state_shape,
+                              np.zeros(shape=state_shape).flatten())
+    lstm_input_names.append(initial_c_name)
+
+    # P (optional) : No peep hole in keras.
+    lstm_input_names.append('')
+
+    activation_types = []
+    alphas = []
+    betas = []
+    extracted_activations = [
+        extract_recurrent_activation(forward_layer.recurrent_activation),
+        extract_recurrent_activation(forward_layer.activation),
+        extract_recurrent_activation(forward_layer.activation),
+        extract_recurrent_activation(backward_layer.recurrent_activation),
+        extract_recurrent_activation(backward_layer.activation),
+        extract_recurrent_activation(backward_layer.activation)]
+
+    for (activation_type, alpha, beta) in extracted_activations:
+        activation_types.append(activation_type.encode('ascii'))
+        if alpha is not None:
+            alphas.append(alpha)
+        if beta is not None:
+            betas.append(beta)
+
+    lstm_attrs['activations'] = activation_types
+    if alphas:
+        lstm_attrs['activation_alpha'] = alphas
+    if betas:
+        lstm_attrs['activation_beta'] = betas
+
+    lstm_attrs['direction'] = 'bidirectional'
+    lstm_attrs['hidden_size'] = hidden_size
+    if operator.targeted_onnx_version < StrictVersion('1.2'):
+        # This attribute exists only before ONNX-1.2
+        lstm_attrs['output_sequence'] = 1 if output_seq else 0
+        op_version = 1
+    else:
+        op_version = 7
```

Comment:
shall we throw an exception here and let the user know lstmlayer.output_seq can't be converted on ONNX 1.2?

Reply:
The attribute output_sequence is not a flag we need to convert directly. If output_sequence=True, we create Keras output from the Y output in ONNX LSTM. If output_sequence=False, the output is produced from Y_h in ONNX LSTM.